### DISCUSSION TITLE
Can I contribute to `tokio::task` for `wasm32-unknown-unknown` target?

### DISCUSSION THREAD

**Discussion by temeddix** [tokio-rs/tokio#6093] [Ideas] [answered]
Hello, I'm the author of [async-wasm-task](https://crates.io/crates/async_wasm_task) which mimics `tokio::task` functionalities on the web. It's working well as expected, thanks to a few core principles:

- Tokio runtime becomes Javascript async environment(event loop)
- Rust `Future`s get converted to Javascript `Promise`s
- `spawn_blocking`'s external thread pool becomes a pool of web workers(and have a fixed limit of 512, each of them being terminated if idle for 10 seconds)

May I create PRs and integrate these features if it's acceptable? If I can, I will make sure there are clear documentation about the difference from native runtime environment.

**Comment by carllerche** [comment, marked as answer]
Looks nice. Thanks for the offer. Generally, we are for adding more platform support to Tokio. I would suggest approaching it with PRs that are as small as possible but still make sense.
